src/Compilers/Core/Portable/Compilation/SemanticModelProvider.cs

```diff
@@ -0,0 +1,13 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#nullable enable
+
+namespace Microsoft.CodeAnalysis
+{
+    internal abstract class SemanticModelProvider
```

Comment:
>SemanticModelProvider [](start = 28, length = 21)

Consider adding doc comment. Is there a common usage pattern or main consumer?

Reply:
Will add doc comment, the primary consumer right now is analyzer driver or any place that was caching semantic models for tree for known performance reasons (for example SuppressMessageAttributeState.IsDiagnosticSuppressed)